src/Compilers/CSharp/Test/Symbol/Symbols/StaticAbstractMembersInInterfacesTests.cs

```diff
@@ -25088,5 +25088,263 @@ class Test
                 Diagnostic((op == "explicit" ? ErrorCode.ERR_NoExplicitConv : ErrorCode.ERR_NoImplicitConv), (needCast ? "(T)" : "") + "x").WithArguments("int", "T").WithLocation(15, 16)
                 );
         }
+
+        [Fact]
+        [WorkItem(53802, "https://github.com/dotnet/roslyn/issues/53802")]
+        public void TestAmbiguousImplementationMethod_01()
+        {
+            var source1 = @"
+public interface Interface<T, U>
+{
+    abstract static void Method(int i);
+    abstract static void Method(T i);
+    abstract static void Method(U i);
+}
+
+public class Base<T> : Interface<T, T>
+{
+    public static void Method(int i) { }
+    public static void Method(T i) { }
+}
+
+public class Derived : Base<int>, Interface<int, int>
+{
+}
+
+class YetAnother : Interface<int, int>
+{
+    public static void Method(int i) { }
+}
+";
+
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.RegularPreview,
+                                                 targetFramework: TargetFramework.NetCoreApp);
+
+            CompileAndVerify(compilation1, sourceSymbolValidator: validate, symbolValidator: validate, verify: Verification.Skipped).VerifyDiagnostics();
+
+            void validate(ModuleSymbol module)
+            {
+                var b = module.GlobalNamespace.GetTypeMember("Base");
+                var bI = b.Interfaces().Single();
+                var biMethods = bI.GetMembers();
+
+                Assert.Equal("void Interface<T, U>.Method(System.Int32 i)", biMethods[0].OriginalDefinition.ToTestDisplayString());
+                Assert.Equal("void Interface<T, U>.Method(T i)", biMethods[1].OriginalDefinition.ToTestDisplayString());
+                Assert.Equal("void Interface<T, U>.Method(U i)", biMethods[2].OriginalDefinition.ToTestDisplayString());
+
+                var bM1 = b.FindImplementationForInterfaceMember(biMethods[0]);
+
+                Assert.Equal("void Base<T>.Method(System.Int32 i)", bM1.ToTestDisplayString());
+
+                var bM2 = b.FindImplementationForInterfaceMember(biMethods[1]);
+
+                Assert.Equal("void Base<T>.Method(T i)", bM2.ToTestDisplayString());
+                Assert.Same(bM2, b.FindImplementationForInterfaceMember(biMethods[2]));
+
+                var bM1Impl = ((MethodSymbol)bM1).ExplicitInterfaceImplementations;
+                var bM2Impl = ((MethodSymbol)bM2).ExplicitInterfaceImplementations;
+
+                if (module is PEModuleSymbol)
+                {
+                    Assert.Equal(biMethods[0], bM1Impl.Single());
```

Comment:
Should this be `Assert.Same`? (also for asserts on `biMethods[1]`, `biMethods[2]` below)

Reply:
> Should this be Assert.Same? (also for asserts on biMethods[1], biMethods[2] below)

When we are dealing with generic instantiations, Assert.Same doesn't work. The symbols are not unique, but often equal and represent the same entity.